install.sh

```diff
@@ -209,86 +196,202 @@ fi
+
+  # provide logging properties
+  # TODO check integration of logging.properties with recent versions of solr
+  ln -sf "$BASEDIR/solrconfig/logging.properties" opus4/logging.properties
+
+  # detect URL prefix to use
+  case "$SOLR_MAJOR" in
+    5)
+      SOLR_CONTEXT="/solr/solr"
+      ;;
+    *)
+      SOLR_CONTEXT="/solr"
+  esac
+
+
+
+  # write solr-config to application's config.ini
+  CONFIG_INI="$BASEDIR/opus4/application/configs/config.ini"
+  "$SCRIPT_PATH/install-config-solr.sh" "$CONFIG_INI" localhost "$SOLR_SERVER_PORT" "${SOLR_CONTEXT}" localhost "$SOLR_SERVER_PORT" "${SOLR_CONTEXT}"
+
   # change file owner of solr installation
-  chown -R "$OWNER" "$BASEDIR/apache-solr-1.4.1"
+  chown -R "$OWNER" "$BASEDIR/$SOLR_DIR"
   chown -R "$OWNER" "$BASEDIR/solrconfig"
 
-  # start Solr server
-  ./opus4-solr-jetty start
+  # install init script
+  [ -z "$INSTALL_INIT_SCRIPT" ] && read -p "Install init.d script to start and stop Solr server automatically? [Y]: " INSTALL_INIT_SCRIPT
+  if [ -z "$INSTALL_INIT_SCRIPT" ] || [ "$INSTALL_INIT_SCRIPT" = Y ] || [ "$INSTALL_INIT_SCRIPT" = y ]
+  then
+    # remove files and folders causing unneccessary errors in install script
+    rm -f /etc/init.d/{opus4-solr-jetty,solr}
+    rm -f "${BASEDIR}/solr"
+
+    # run installer bundled with solr
+    bin/install_solr_service.sh "../downloads/$SOLR_ARCHIVE_NAME" -d "$SOLR_CORE_DIR" -i "$BASEDIR" -p "$SOLR_SERVER_PORT" -s solr -u "$OPUS_USER_NAME"
+
+    # make sure new service is available just like the old one
+    ln -s solr /etc/init.d/opus4-solr-jetty
+  else
+    # (re)start solr service
+    if [ -x /etc/init.d/opus4-solr-jetty ]; then
+      /etc/init.d/opus4-solr-jetty restart
+    elif [ -x /etc/init.d/solr ]; then
+      /etc/init.d/solr restart
+    fi
+  fi
 fi
 
 # import some test documents
-[[ -z $IMPORT_TESTDATA ]] && read -p "Import test data? [Y]: " IMPORT_TESTDATA
+[ -z "$IMPORT_TESTDATA" ] && read -p "Import test data? [Y]: " IMPORT_TESTDATA
 if [ -z "$IMPORT_TESTDATA" ] || [ "$IMPORT_TESTDATA" = Y ] || [ "$IMPORT_TESTDATA" = y ]
 then
   # import test data
   cd "$BASEDIR"
   for i in `find opus4/tests/sql -name *.sql \( -type f -o -type l \) | sort`; do
     echo "Inserting file '${i}'"
-    eval "$MYSQL_OPUS4ADMIN" "$DBNAME" < "${i}"
+    mysqlOpus4Admin "$DBNAME" < "${i}"
   done
 
   # copy test fulltexts to workspace directory
   cp -rv opus4/tests/fulltexts/* workspace/files
 
+  # TODO is waiting for running solr required since service script has been waiting for this before
   # sleep some seconds to ensure the server is running
   echo -en "\n\nwait until Solr server is running..."
 
-  while :; do
+  waiting=true
+
+  pingSolr() {
+    wget -SO- "$1" 2>&1
+  }
+
+  pingSolrStatus() {
+    pingSolr "$1" | sed -ne 's/^ *HTTP\/1\.[01] \([0-9]\+\) .\+$/\1/p' | head -1
+  }
+
+  case "$SOLR_MAJOR" in
+    5)
+      PING_URL="http://localhost:${SOLR_SERVER_PORT}${SOLR_CONTEXT}/admin/ping"
```

Comment:
Revision leftover ;) ... it became dead code actually on introducing SOLR_CONTEXT not in use before.